### PR TITLE
Refactor: Rename some parameters for clarity

### PR DESCRIPTION
**Pull request by Venn1998**
Update the formula for Gradient Descent in the introduction to be consistent with the slides and the formula in the first exercise.

Changed the name of some parameters in formulae to be less confusing as feedback suggested, and added some comments for the same reason. 

Fixed minor typos in text.

### CODE REVIEW

**Comment by Lovkush-A** [comment]
looks great. one small idea is to remove the exercise in the `optimize_function` function.  It is surprisingly tricky to work out what is expected here, and they will have opportunity to write this loop in hyper-params notebook / pre-work.

Explicitly, change:
```python
        # Hide: all
        loss = function(*parameters)
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
        # Hide: none
```

to 

```python
        loss = function(*parameters)
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
```
